### DISCUSSION TITLE
Delay/Minimum time in state

### DISCUSSION THREAD

**Discussion by douglaseggleton** [statelyai/xstate#2328] [Q&A]
I've been looking at the delay documentation...
https://xstate.js.org/docs/guides/delays.html#delayed-transitions 🙌🏻

I want to set a 'minimum time in state' of 2 seconds before any transition can occur.
If an event occurs at 1s, it should wait until the 2s has passed before the transition happens. 
If an event at 3s should transition immediately.

If anyone can point me in the right direction for this behaviour, much appreciated 🤗

(P.s. XState is awesome 💯 )

**Comment by davidkpiano** [comment]
You can do this explicitly by using a combination of nested states and [defer/recalled events](https://github.com/davidkpiano/xstate/discussions/1305#discussioncomment-37644) (not shown in the below code)

Here is how you would prevent a transition from happening before a certain time:

```js
// ...
states: {
  normal: {
    initial: 'waiting',
    states: {
      waiting: {
        on: { '*': {/* here is where you would defer */} },
        after: { 2000: 'ready' }
      },
      ready: {
        entry: {/* here is where you would recall deferred events */}
        // all your transitions
      }
    }
  }
}
```

**Reply by douglaseggleton** [reply]
Thanks for your help. Worked a treat 👌🏻
